source/electricity-market/ElectricityMarket.Application/Services/CapacitySettlementService.cs

```
@@ -0,0 +1,155 @@
+using NodaTime;
+
+namespace Energinet.DataHub.ElectricityMarket.Application.Services;
+
+public class CapacitySettlementService : ICapacitySettlementService
+{
+    private readonly IMeteringPointRepository _meteringPointRepository;
+    private readonly Instant _capacitySettlementEnabledFrom = Instant.FromUtc(2025, 1, 1, 0, 0, 0);
+
+    public CapacitySettlementService(IMeteringPointRepository meteringPointRepository)
+    {
+        _meteringPointRepository = meteringPointRepository;
+    }
+
+    // """
+    // Metering point periods for consumption metering points (parent) that have a coupled 'capacity_settlement' metering point (child).
+    //
+    // It represents the timeline of the consumption metering points. The first period (given by period_from_date/period_from_to)
+    // of each metering point starts when the metering point first time enters 'connected' state - or 'disconnected' if that
+    // occurs first. After that, new period starts when (and only when) a 'move-in' occurs, and the previous period is then
+    // terminated at that same time.
+    //
+    // Exclude rows where the period of the parent
+    // - does not have any overlap with the period of the child metering point.
+    // - ends before 2024-12-31 23:00:00
+    //
+    // Formatting is according to ADR-144 with the following constraints:
+    // - No column may use quoted values
+    // - All date/time values must include seconds
+    // """
+    public async IAsyncEnumerable<ICapacitySettlementResult> GetCapacitySettlementPeriodsAsync(
+        MeteringPointIdentification meteringPointIdentification, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        var meteringPointHierarchy = await _meteringPointRepository.GetMeteringPointHierarchyAsync(meteringPointIdentification, cancellationToken).ConfigureAwait(false);
+
+        var capacitySettlementMeteringPoint =
+            meteringPointHierarchy.ChildMeteringPoints.SingleOrDefault(mp =>
```

can there only 1 with this criteria?

---

According to Irene there can be only one D19 per E17

---

ok, but what if the parent is not E17? can that be?

---

So you are considering the case where a Non-E17 parent metering point has multiple D19 child metering points?
I have no idea, but maybe we should let this fail fast for now, and the handle later in the full implementation since QA may provide some of these examples?
(For now this will explode if there are multiple D19 child metering points, and I don't know how to handle multiple :-))